coriolis/osmorphing/base.py

```diff
@@ -427,6 +428,49 @@ def _configure_cloud_init_user_retention(self):
                     "Failed to reconfigure cloud-init to retain user "
                     "credentials. Error was: %s" % str(err)) from err
 
+    def _disable_installer_cloud_config(self):
+        installer_config_path = "/etc/cloud/cloud.cfg.d/99-installer.cfg"
+        if self._test_path(installer_config_path):
+            self._event_manager.progress_update(
+                "Disabling Ubuntu installer-generated cloud-config")
+            self._exec_cmd_chroot(
+                f"mv {installer_config_path} {installer_config_path}.bak")
+
+    def _ensure_cloud_init_enabled(self):
+        disabler_file = "/etc/cloud/cloud-init.disabled"
+        if self._test_path(disabler_file):
+            self._exec_cmd_chroot(f"rm {disabler_file}")
+        # self._exec_cmd_chroot("systemctl enable cloud-init")
+
+    def _reset_cloud_init_run(self):
+        self._exec_cmd_chroot("cloud-init clean --logs")
+
+    def _get_default_cloud_user(self):
+        cloud_cfg_path = 'etc/cloud/cloud.cfg'
+        if not self._test_path(cloud_cfg_path):
+            return DEFAULT_CLOUD_USER
+        cloud_cfg_content = self._read_file(cloud_cfg_path)
+        cloud_cfg = yaml.load(cloud_cfg_content, Loader=yaml.SafeLoader)
+        return cloud_cfg.get('system_info', {}).get('default_user', {}).get(
+            'name', DEFAULT_CLOUD_USER)
+
+    def _create_cloudinit_user(self):
+        cloud_user = self._get_default_cloud_user()
+        if not self._check_user_exists(cloud_user):
+            self._exec_cmd_chroot("useradd %s" % cloud_user)
+
+    def _configure_cloud_init(self):
+        if "cloud-init" not in self.get_packages()[0]:
+            return
+
+        self._disable_installer_cloud_config()
+        self._ensure_cloud_init_enabled()
+        self._reset_cloud_init_run()
+        if self._osmorphing_parameters.get('retain_user_credentials', False):
+            self._configure_cloud_init_user_retention()
```

Comment:
This method overwrites cloud.cfg. I'd like to add those mods into a separate file called `99_coriolis.cfg` inside `cloud.cfg.d`. Save them inside a `cloud_cfg_mods` and only at the end of this `_configure_cloud_init` method you'd write that file. 
Also, here is what the `set_dhcp` mods should also be included.